app/src/main/java/org/schabi/newpipe/util/DeviceUtils.java

```diff
@@ -70,4 +71,15 @@ public static boolean isConfirmKey(final int keyCode) {
                 return false;
         }
     }
+
+    public static int dpToPx(final float dp, final Context context) {
+        return (int) (dp * context.getResources().getDisplayMetrics().density);
+    }
+
+    public static int spToPx(final float sp, final Context context) {
+        return (int) TypedValue.applyDimension(
+                TypedValue.COMPLEX_UNIT_SP,
+                sp,
+                context.getResources().getDisplayMetrics());
+    }
```

Comment:
Tests would be nice

Reply:
`spToPx` is the recommended way to do it, I will also change the `dpToPx` method to do the same. These methods are used in various internal android libs (e.g. see `com.google.android.material.internal.ViewUtils.dpToPx(Context context, int dp)`), so I do not think they need tests.